core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```
@@ -545,14 +545,19 @@ object ConsumerGroupCommand extends Logging {
         .map(partitionInfo => new TopicPartition(topic, partitionInfo.partition))
     }
 
-    private def getPartitionsToReset(groupId: String): Iterable[TopicPartition] = {
+    private def getPartitionsToReset(groupId: String): Seq[TopicPartition] = {
+      val allTopicPartitions = adminClient.listGroupOffsets(groupId).keys.toSeq
       if (opts.options.has(opts.allTopicsOpt)) {
-        adminClient.listGroupOffsets(groupId).keys
+        allTopicPartitions
       } else if (opts.options.has(opts.topicOpt)) {
         val topics = opts.options.valuesOf(opts.topicOpt).asScala
-        parseTopicPartitionsToReset(topics)
+        val topicPartitions = parseTopicPartitionsToReset(topics)
+        allTopicPartitions.filter(topicPartition => topicPartitions.contains(topicPartition))
```

I'm not sure I get why we filter based on current committed offsets. Why would we not permit the tool to set offsets for a topic partition which hasn't been previously consumed? If that is really the behavior we want, I would prefer to raise an exception rather than silently proceeding with only some of the offsets.

---

I haven't considered setting offsets on topic partitions unassigned previously, sounds like a good idea (e.g. set initial offsets for a new consumer group). if we add this, checking previous assignments won't be required, right? And implementation will be cleaner. Maybe we can add this as part of this PR?

---

Yes, I think it is a reasonable feature to implement here and users might find it a little surprising if they couldn't do it.